### ISSUE TITLE
RFE: Skill Specializations and Natural Aptitudes

### ISSUE THREAD

**Issue by Saklad5** [MegaMek/mekhq#4428]
MekHQ currently has a relative dearth of official things for support personnel to spend XP on besides Skill Ratings. It also has a few unofficial things that are similar to but do not follow official *A Time of War* character options.

I think it is worth polishing this to reflect official rules where possible, both for the sake of balance and to provide players using MekHQ alongside AToW a better experience.

# Official Rules

## Skill Specializations

In *A Time of War*, any subskill (or skill, if it has no subskills) can be given a specialization.[^1] The specific options are deliberately left to a gamemaster, though in MekHQ's case that's obviously up to us.

The rules are always the same, though: a specialty is effectively free, only one can be chosen for each subskill, and it provides +1 when applicable[^2]. However, when it is _not_ applicable, the character suffers -1.

MekHQ already implements specializations to some degree, though in the form of specific SPAs like "Gunnery Specialization". It also has unofficial skills that are very close to following the official rules, like "Tech Specialist". The main divergences right now are that both should be tied to a Skill Rating rather than an SPA, and both should penalize unspecialized uses of that skill.

## Natural Aptitude

While also currently implemented as an SPA, Natural Aptitude is actually a Trait that applies to a specific subskill/skill, in a similar manner to specializations. It is currently in MekHQ for "Gunnery" and "Piloting".

# Proposed Change

In the interest of following official rules, reducing interface clutter, and expanding what MegaMek is capable of, both specializations and the Natural Aptitude trait should be removed from the list of Special Pilot Abilities. Instead they should be placed alongside each Skill Rating for an individual, in the form of a dropdown for a specialization and a checkbox for Natural Aptitude.

The specific specializations available aren't mandated, though there are examples given for most of the currently-implemented skills:

* Admin (HR)
* Gunnery/'Mech (Energy)
* Small Arms (Shotguns)
* Technician/Mechanics (Fuel Cell Engines)[^2]

[^1]: Page 141, *A Time of War*, third printing
[^2]: *A Time of War* provides more detailed Technician subskills than *Campaign Operations*, and MekHQ currently implements the latter. This could actually make specialization a lot less powerful. For example, a MechTech in AToW could take "Technician/Nuclear (Fusion Engines)" without penalizing much besides fission engines, and at the same time take "Technician/Myomer (Actuators)" and suffer mostly with MASC or AES: both have fairly limited applications with regard to BattleMechs.

**Comment by RaozSpaz** [comment]
The various +1/-1 for the 3 big damage categories definitely already exist. 
The small arms ones while neat in theory would never outweigh the usefulness of just getting weapon specialist instead.
The technician ones I dont think would ever really see much use outside of people who always micromanage their repairs. Cause a +1/-1 skill for a skill with that much variable use is SUPER niche. 
The admin skill is the only one I could possibly see it being used as Admins are generally pretty specific in their job function.

**Comment by Saklad5** [comment]
> The technician ones I dont think would ever really see much use outside of people who always micromanage their repairs. Cause a +1/-1 skill for a skill with that much variable use is SUPER niche.

There's currently a set of unofficial SPAs called "Tech Specialist" that is basically the same thing. However, since MekHQ is using the broader CamOps Technician skills, we might want to make the specializations broader as well.

**Comment by RaozSpaz** [comment]
I mean the only thing broader would be adding something like equipment. Which at a general glance is the only thing not covered by the 3 listed in MHQ. But the big difference between those and what you listed is they are just a flat -1 compared to the trade off that you listed. Which would rarely if ever see use in a field so widespread as technician work.

**Comment by Saklad5** [comment]
Personally, I'd gladly have someone specialized in something super-difficult.

**Comment by RaozSpaz** [comment]
Yes, but unless you fully remove the unofficial ones [or let the unofficial ones stack with your additions], there is no value to taking your proposed additions over the ones already implemented.
